cmd/juju/machine/add.go

```diff
@@ -341,3 +364,32 @@ func (c *addCommand) Run(ctx *cmd.Context) error {
 	}
 	return nil
 }
+
+func (c *addCommand) scopeRun(client AddMachineAPI, config *config.Config, ctx *cmd.Context) error {
```

Comment:
I think you should move all of the code from "manualProvisioner" into here, so it looks like:

```go

var sshprovisionerProvisionMachine = sshprovisioner.ProvisionMachine
var errNonManualScope = errors.New("non-manual scope")

func (c *addCommand) maybeManuallyProvisionMachine(client AddMachineAPI, config *config.Config, ctx *cmd.Context) error {
    var provisionMachine func(manual.ProvisionMachineArgs) (string, error)
    switch c.Placement.Scope {
    case "ssh":
        manualProvisioner = sshprovisionerProvisionMachine
    default:
        return errNonManualScope
    }

    logger.Infof("manual provisioning (%q)", c.Placement.Scope)
    authKeys, err ...
    user, host ...
    args := ...

    machineId, err := provisionMachine(args)
    ...
}
```